src/Microsoft.OpenApi/Services/OpenApiFilterService.cs

```diff
@@ -46,16 +56,52 @@ public static class OpenApiFilterService
                 {
                     var regex = new Regex(tagsArray[0]);
 
-                    predicate = (o) => o.Tags.Any(t => regex.IsMatch(t.Name));
+                    predicate = (url, operationType, o) => o.Tags.Any(tag => regex.IsMatch(tag.Name));
                 }
                 else
                 {
-                    predicate = (o) => o.Tags.Any(t => tagsArray.Contains(t.Name));
+                    predicate = (url, operationType, o) => o.Tags.Any(tag => tagsArray.Contains(tag.Name));
```

Comment:
Since we are choosing to give the predicate args meaningful names, why don't we also rename `o` to `operation`.